src/PaginationCommand.php

```diff
@@ -7,63 +7,87 @@
+            ->addOption("filter", "f", InputOption::VALUE_OPTIONAL, 'page size', '');
     }
 
     protected
-    abstract function get_endpoint();
+    abstract
+    function get_endpoint();
 
-    protected
-    function execute(InputInterface $input, OutputInterface $output) {
+    public
+    function getResult(InputInterface $input, OutputInterface $output) {
         try {
-            $page_num = 1;
-            $client = $this->getClient();
-            $page_size = $input->getOption('size');
+            $items = new Sequence();
+
+            $page_num     = 1;
+            $client       = $this->getClient();
+            $page_size    = $input->getOption('size');
             $page_num_max = $input->getOption('pages');
+            $filter       = $input->getOption('filter');
 
             if ($page_size > 25) {
                 $output->writeln("$page_size exceeds page size limit 25, setting it to 25 ...");
                 $page_size = 25;
             }
 
             $url = $this->get_endpoint();
-            $r = $client->get($url, ['query' => ['limit' => $page_size]]);
+            $r   = $client->get($url, [
+                'query'               => [
+                    'limit'  => $page_size,
+                    'filter' => $filter
+                ],
+                RequestOptions::DEBUG =>
+                    boolval($input->getOption('debug')) ? fopen('php://stdout',
+                                                                'w') : false
+            ]);
+
             $result = json_decode($r->getBody()->getContents());
-            $this->printResult($page_num, $result, $output);
+            $items->addAll($result->items);
+
 
             while (isset($result->links) && !empty($result->links->next) && $page_num < $page_num_max) {
                 $url = $result->links->next;
-                $r = $client->get($url);
+                $r   = $client->get($url);
                 $page_num++;
                 $result = json_decode($r->getBody()->getContents());
-                $this->printResult($page_num, $result, $output);
+                $items->addAll($result->items);
             }
+
+            return $items->all();
         }
         catch (ClientException $e) {
             $response = $e->getResponse();
             $body     = $response->getBody()->getContents();
             $output->writeln("Failed to grab token: " . $body);
             throw $e;
         }
+
     }
 
-    private
-    function printResult($page_num, $result, OutputInterface $output) {
-        $output->writeln("Loading page $page_num of results");
-        $output->writeln(print_r($result,true));
-        $output->writeln("----------------");
+
+    protected
+    function execute(InputInterface $input, OutputInterface $output) {
+        $items = $this->getResult($input, $output);
+
+        $output->writeln(json_encode($items, JSON_PRETTY_PRINT));
```

Comment:
yes.   So this allows you to do a ./bin/imc list:somestuff | json_query ./blahblah